kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/PodOperationsImpl.java

```diff
@@ -262,9 +257,10 @@ public ExecWatch exec(String... command) {
         try {
             URL url = new URL(URLUtils.join(getResourceUrl().toString(), sb.toString()));
             Request.Builder r = new Request.Builder().url(url).header("Sec-WebSocket-Protocol", "v4.channel.k8s.io").get();
-            OkHttpClient clone = client.newBuilder().readTimeout(0, TimeUnit.MILLISECONDS).build();
             final ExecWebSocketListener execWebSocketListener = new ExecWebSocketListener(getConfig(), in, out, err, errChannel, inPipe, outPipe, errPipe, errChannelPipe, execListener);
-            clone.newWebSocket(r.build(), execWebSocketListener);
+            Config requestConfig = new ConfigBuilder(config).withRequestTimeout(0).build();
```

Comment:
are you sure we want top drop clonned client?

Reply:
@hrishin : I'm not sure whether dropping client is right thing here but it's the only way from which I was able to reduce leaks.